java-checks/src/main/java/org/sonar/java/checks/StringLiteralDuplicatedCheck.java

```diff
@@ -90,13 +92,58 @@ private static List<JavaFileScannerContext.Location> secondaryLocations(Collecti
   public void visitLiteral(LiteralTree tree) {
     if (tree.is(Tree.Kind.STRING_LITERAL, Tree.Kind.TEXT_BLOCK)) {
       String literal = tree.value();
-      if (literal.length() >= MINIMAL_LITERAL_LENGTH) {
+      if (literal.length() >= MINIMAL_LITERAL_LENGTH && !isStringLiteralFragment(tree)) {
         String stringValue = LiteralUtils.getAsStringValue(tree).replace("\\n", "\n");
         occurrences.computeIfAbsent(stringValue, key -> new ArrayList<>()).add(tree);
       }
     }
   }
 
+  private static boolean isStringLiteralFragment(ExpressionTree tree) {
+    return isStringLiteral(tree) && (
+      isStringLiteral(getNextOperand(tree, Tree.Kind.PLUS)) ||
+        isStringLiteral(getPreviousOperand(tree, Tree.Kind.PLUS))
+    );
+  }
+
+  private static boolean isStringLiteral(@Nullable Tree tree) {
+    return tree != null && tree.is(Tree.Kind.STRING_LITERAL);
+  }
+
+  @Nullable
+  private static ExpressionTree getNextOperand(ExpressionTree tree, Tree.Kind kind) {
+    var binary = asBinaryExpression(tree.parent(), kind);
+    if (binary == null) {
+      return null;
+    }
+    if (tree == binary.leftOperand()) {
+      return binary.rightOperand();
+    } else {
+      binary = asBinaryExpression(binary.parent(), kind);
+      return binary != null? binary.rightOperand(): null;
+    }
+  }
+
+  @Nullable
+  private static ExpressionTree getPreviousOperand(ExpressionTree tree, Tree.Kind kind) {
+    var binary = asBinaryExpression(tree.parent(), kind);
+    if (binary == null) {
+      return null;
+    }
+    if (tree == binary.leftOperand()) {
+      return null;
+    } else {
+      var left = binary.leftOperand();
+      binary = asBinaryExpression(left, kind);
+      return binary != null? binary.rightOperand(): binary;
+    }
+  }
+
+  @Nullable
+  private static BinaryExpressionTree asBinaryExpression(Tree tree, Tree.Kind kind) {
+    return tree.is(kind)? (BinaryExpressionTree) tree: null;
```

Comment:
Since we are only dealing with strings here, we will never need to use other kinds than PLUS, so we can remove the method's argument and adapt the other method calls in the check
```suggestion
  private static BinaryExpressionTree asBinaryExpression(Tree tree) {
    return tree.is(Tree.Kind.PLUS) ? (BinaryExpressionTree) tree: null;
```